File: AspNetMvcUnleashed/0672329980_source/Chapter04/CS/MvcApplication1/MvcApplication1.Tests/Controllers/HomeControllerTest.cs
using System.Web.Mvc;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using MvcApplication1.Controllers;

namespace MvcApplication1.Tests.Controllers
{
    [TestClass]
    public class HomeControllerTest
    {
        [TestMethod]
        public void Index()
        {
            // Arrange
            var controller = new HomeController();
 
            // Act
            var result = controller.Index();

            // Did we get a view result?
            Assert.IsInstanceOfType(result, typeof(ViewResult));

            // Did we get a view named Index?
            var indexResult = (ViewResult)result;
            Assert.AreEqual("Index", indexResult.ViewName);

            // Did we get message in view data?
            Assert.AreEqual("Hello World!", indexResult.ViewData["message"]);
        }

    }
}

File: AspNetMvcUnleashed/0672329980_source/Chapter04/VB/MvcApplication1/MvcApplication1.Tests/Controllers/HomeControllerTest.vb
Imports Microsoft.VisualStudio.TestTools.UnitTesting
Imports System.Web.Mvc

<TestClass()> _
Public Class HomeControllerTest
    <TestMethod()> _
    Public Sub Index()
        ' Arrange
        Dim controller = New HomeController()

        ' Act
        Dim result = controller.Index()

        ' Did we get a view result?
        Assert.IsInstanceOfType(result, GetType(ViewResult))

        ' Did we get a view named Index?
        Dim indexResult = CType(result, ViewResult)
        Assert.AreEqual("Index", indexResult.ViewName)

        ' Did we get message in view data?
        Assert.AreEqual("Hello World!", indexResult.ViewData("message"))
    End Sub

End Class
